packages/nova-lib/lib/client/render_context.js

```diff
@@ -1,56 +1,51 @@
-import { createStore, applyMiddleware, compose, combineReducers } from 'redux';
 import { browserHistory } from 'react-router';
+import { compose } from 'redux';
 
-import { createApolloClient, addReducer, addMiddleware, getReducers, getMiddlewares } from '../modules/index.js';
+import {
+  createApolloClient,
+  configureStore, STORE_RELOADED,
+  addAction, getActions, addReducer, getReducers, addMiddleware, getMiddlewares,
+  Utils,
+} from '../modules/index.js';
 
+// init
 const history = browserHistory;
-
 const loginToken = global.localStorage['Meteor.loginToken'];
 const apolloClient = createApolloClient();
+addReducer({ apollo: apolloClient.reducer() });
+addMiddleware(Utils.defineName(apolloClient.middleware(), 'apolloClientMiddleware'));
 
-const reducers = addReducer({ apollo: apolloClient.reducer() });
-const middlewares = addMiddleware(apolloClient.middleware());
-
-const STORE_RELOADED = 'STORE_RELOADED';
+// init context
+const context = {
+  history,
+  loginToken,
+  apolloClient,
+  addAction, // context.addAction same as addAction
```

Comment:
just want to keep the client side context's function same as server side context's function
on server side.
addAction for add an action to server's actions array
context.addAction for add an action to context's actions array
and then when SSR a request come
the final actions = server's actions array + context's actions array

so i think keep client side context with the same structure is better.
and yes. it is not necessary